package.json

```diff
@@ -8,9 +8,8 @@
   "main": "index.js",
   "scripts": {
     "precommit": "lint-staged",
-    "localtestsetup": "python3 -m venv test/venv && . test/venv/bin/activate && pip install -r test/requirements.txt",
-    "testlocal": "npm run localtestsetup && npm run test",
-    "test": "python3 ./test/automated_tests.py"
+    "test": "mocha \"test/unit/**/*.test.js\"",
+    "integration-test": "mocha-isolated --pass-through-aws-creds \"test/integration/**/*.test.js\""
```

Comment:
Whenever I run this it fails:

```

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
Shmac:serverless-finch shereef.marzouk$ export AWS_ACCESS_KEY_ID=“*”**************
Shmac:serverless-finch shereef.marzouk$ export AWS_SECRET_ACCESS_KEY=“*”***************
Shmac:serverless-finch shereef.marzouk$ export AWS_SESSION_TOKEN=“*”********************
Shmac:serverless-finch shereef.marzouk$ npm run integration-test

> serverless-finch@2.8.0 integration-test
> mocha-isolated --pass-through-aws-creds "test/integration/**/*.test.js"



  client deploy integration test
    us-east-1
2022-02-25T05:19:25.727Z i test Setup 'serverless' at /var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3
2022-02-25T05:20:38.055Z i serverless:plugin-legacy This deployment will:
2022-02-25T05:20:38.056Z i serverless:plugin-legacy - Upload all files from 'client/dist' to bucket 'sls-finch-test-stndrd-use1-v3' undefined
2022-02-25T05:20:38.056Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' configuration
2022-02-25T05:20:38.056Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' bucket policy
2022-02-25T05:20:38.056Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' CORS policy
2022-02-25T05:20:38.056Z i serverless:plugin-legacy Looking for bucket...
      1) should deploy site sucessfully
2022-02-25T05:20:38.286Z × mocha test fail ServerlessError: ServerlessError: The AWS Access Key Id you provided does not exist in our records.\n    at /Users/shereef.marzouk/Dev/serverless-finch/index.js:314:31\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at async PluginManager.runHooks (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:522:35)\n    at async PluginManager.invoke (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:552:9)\n    at async PluginManager.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:592:7)\n    at async Serverless.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/serverless.js:169:5)\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:366:15\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:235:28\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/setup-run-serverless-fixtures-engine.js:33:22\n    at async Context.<anonymous> (test/integration/deploy.test.js:22:9)
2022-02-25T05:20:38.287Z i mocha flushing previously gathered logs...
2022-02-25T05:19:25.704Z * mocha 
2022-02-25T05:19:25.705Z * mocha client deploy integration test
2022-02-25T05:19:25.706Z * mocha client deploy integration test > us-east-1
2022-02-25T05:19:25.706Z * mocha client deploy integration test > us-east-1 > should deploy site sucessfully
2022-02-25T05:19:25.726Z i serverless:test setup basic fixture at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/tmpdirs-serverless/cd1c/ce4f16
2022-02-25T05:19:25.727Z i test Setup 'serverless' at /var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3
2022-02-25T05:19:25.727Z * test ... fetch tarball
2022-02-25T05:19:27.133Z * test ... patch serverless/package.json
2022-02-25T05:19:27.139Z * child-process-ext:spawn [1] run npm with [ 'install', '--production', [length]: 2 ]
2022-02-25T05:20:22.883Z * child-process-ext:spawn:std:err [1] npm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau\n
2022-02-25T05:20:24.181Z * child-process-ext:spawn:std:err [1] npm WARN
2022-02-25T05:20:24.181Z * child-process-ext:spawn:std:err [1]  deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.\n
2022-02-25T05:20:24.739Z * child-process-ext:spawn:std:err [1] npm
2022-02-25T05:20:24.739Z * child-process-ext:spawn:std:err [1]  WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.\n
2022-02-25T05:20:24.812Z * child-process-ext:spawn:std:err [1] npm 
2022-02-25T05:20:24.812Z * child-process-ext:spawn:std:err [1] WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.\n
2022-02-25T05:20:24.845Z * child-process-ext:spawn:std:err [1] npm WARN
2022-02-25T05:20:24.845Z * child-process-ext:spawn:std:err [1]  deprecated superagent@3.8.3: Please upgrade to v7.0.2+ of superagent.  We have fixed numerous issues with streams, form-data, attach(), filesystem errors not bubbling up (ENOENT on attach()), and all tests are now passing.  See the releases tab for more information at <https://github.com/visionmedia/superagent/releases>.\n
2022-02-25T05:20:24.869Z * child-process-ext:spawn:std:err [1] npm WARN 
2022-02-25T05:20:24.869Z * child-process-ext:spawn:std:err [1] deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.\n
2022-02-25T05:20:26.954Z * child-process-ext:spawn:std:out [1] \nadded 384 packages, and audited 385 packages in 60s\n
2022-02-25T05:20:26.954Z * child-process-ext:spawn:std:out [1] \n
2022-02-25T05:20:26.954Z * child-process-ext:spawn:std:out [1] 46 packages are looking for funding\n  run `npm fund` for details\n
2022-02-25T05:20:26.955Z * child-process-ext:spawn:std:out [1] \nfound 0 vulnerabilities\n
2022-02-25T05:20:26.956Z * child-process-ext:spawn:std:err [1] npm notice \n
2022-02-25T05:20:26.956Z * child-process-ext:spawn:std:err [1] npm notice New minor version of npm available! 8.4.0 -> 8.5.2\nnpm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.5.2>\nnpm notice Run `npm install -g npm@8.5.2` to update!\nnpm notice \n
2022-02-25T05:20:26.992Z * child-process-ext:spawn [1] succeeded
2022-02-25T05:20:26.992Z * test ... link serverless-finch dependency
2022-02-25T05:20:30.956Z i serverless To ensure safe major version upgrades ensure "frameworkVersion" setting in service configuration (recommended setup: "frameworkVersion: ^3.3.0")

2022-02-25T05:20:36.247Z * serverless:lifecycle:command:register config
2022-02-25T05:20:36.248Z * serverless:lifecycle:command:register config:credentials
2022-02-25T05:20:36.250Z * serverless:lifecycle:command:register create
2022-02-25T05:20:36.251Z * serverless:lifecycle:command:register install
2022-02-25T05:20:36.252Z * serverless:lifecycle:command:register package
2022-02-25T05:20:36.252Z * serverless:lifecycle:command:register package:function
2022-02-25T05:20:36.253Z * serverless:lifecycle:command:register deploy
2022-02-25T05:20:36.253Z * serverless:lifecycle:command:register deploy:function
2022-02-25T05:20:36.253Z * serverless:lifecycle:command:register deploy:list
2022-02-25T05:20:36.253Z * serverless:lifecycle:command:register deploy:list:functions
2022-02-25T05:20:36.254Z * serverless:lifecycle:command:register invoke
2022-02-25T05:20:36.254Z * serverless:lifecycle:command:register invoke:local
2022-02-25T05:20:36.255Z * serverless:lifecycle:command:register info
2022-02-25T05:20:36.255Z * serverless:lifecycle:command:register logs
2022-02-25T05:20:36.255Z * serverless:lifecycle:command:register metrics
2022-02-25T05:20:36.256Z * serverless:lifecycle:command:register print
2022-02-25T05:20:36.256Z * serverless:lifecycle:command:register remove
2022-02-25T05:20:36.256Z * serverless:lifecycle:command:register rollback
2022-02-25T05:20:36.256Z * serverless:lifecycle:command:register rollback:function
2022-02-25T05:20:36.257Z * serverless:lifecycle:command:register slstats
2022-02-25T05:20:36.257Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:36.257Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:36.257Z * serverless:lifecycle:command:register plugin:list
2022-02-25T05:20:36.258Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:36.258Z * serverless:lifecycle:command:register plugin:search
2022-02-25T05:20:36.258Z * serverless:lifecycle:command:register config
2022-02-25T05:20:36.258Z * serverless:lifecycle:command:register config:credentials
2022-02-25T05:20:36.280Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:36.280Z * serverless:lifecycle:command:register aws:common
2022-02-25T05:20:36.281Z * serverless:lifecycle:command:register aws:common:validate
2022-02-25T05:20:36.281Z * serverless:lifecycle:command:register aws:common:cleanupTempDir
2022-02-25T05:20:36.281Z * serverless:lifecycle:command:register aws:common:moveArtifactsToPackage
2022-02-25T05:20:36.281Z * serverless:lifecycle:command:register aws:common:moveArtifactsToTemp
2022-02-25T05:20:36.282Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:36.282Z * serverless:lifecycle:command:register aws:package
2022-02-25T05:20:36.282Z * serverless:lifecycle:command:register aws:package:finalize
2022-02-25T05:20:36.282Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:36.282Z * serverless:lifecycle:command:register aws:deploy
2022-02-25T05:20:36.282Z * serverless:lifecycle:command:register aws:deploy:deploy
2022-02-25T05:20:36.283Z * serverless:lifecycle:command:register aws:deploy:finalize
2022-02-25T05:20:36.283Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:36.283Z * serverless:lifecycle:command:register aws:info
2022-02-25T05:20:36.287Z * serverless:lifecycle:command:register upgrade
2022-02-25T05:20:36.288Z * serverless:lifecycle:command:register uninstall
2022-02-25T05:20:36.619Z * serverless:lifecycle:command:register client
2022-02-25T05:20:36.619Z * serverless:lifecycle:command:register client:deploy
2022-02-25T05:20:36.619Z * serverless:lifecycle:command:register client:remove
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register login
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register logout
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register generate-event
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register test
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register dashboard
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register output
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register output:get
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register output:list
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register param
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register param:get
2022-02-25T05:20:36.620Z * serverless:lifecycle:command:register param:list
2022-02-25T05:20:38.055Z * serverless:lifecycle:command:invoke Invoke client:deploy
2022-02-25T05:20:38.055Z i serverless:plugin-legacy This deployment will:
2022-02-25T05:20:38.056Z i serverless:plugin-legacy - Upload all files from 'client/dist' to bucket 'sls-finch-test-stndrd-use1-v3' undefined
2022-02-25T05:20:38.056Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' configuration
2022-02-25T05:20:38.056Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' bucket policy
2022-02-25T05:20:38.056Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' CORS policy
2022-02-25T05:20:38.056Z i serverless:plugin-legacy Looking for bucket...
2022-02-25T05:20:38.282Z * serverless:aws [AWS s3 403 0.203s 0 retries] listBuckets({})
2022-02-25T05:20:38.283Z * serverless InvalidAccessKeyId: The AWS Access Key Id you provided does not exist in our records.
    at Request.extractError (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/services/s3.js:710:35)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at callNextListener (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:96:12)
    at IncomingMessage.onEnd (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/event_listeners.js:335:13)
    at IncomingMessage.emit (events.js:412:35)
    at IncomingMessage.emit (domain.js:475:12)
    at endReadableNT (internal/streams/readable.js:1334:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
----------------------------------------------------------------------------------------------------
    us-west-2
2022-02-25T05:20:38.419Z i serverless:plugin-legacy This deployment will:
2022-02-25T05:20:38.419Z i serverless:plugin-legacy - Upload all files from 'client/dist' to bucket 'sls-finch-test-stndrd-usw2-fer' undefined
2022-02-25T05:20:38.419Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-usw2-fer' configuration
2022-02-25T05:20:38.419Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-usw2-fer' bucket policy
2022-02-25T05:20:38.419Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-usw2-fer' CORS policy
2022-02-25T05:20:38.419Z i serverless:plugin-legacy Looking for bucket...
      2) should deploy site sucessfully
2022-02-25T05:20:38.899Z × mocha test fail ServerlessError: ServerlessError: The AWS Access Key Id you provided does not exist in our records.\n    at /Users/shereef.marzouk/Dev/serverless-finch/index.js:314:31\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at async PluginManager.runHooks (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:522:35)\n    at async PluginManager.invoke (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:552:9)\n    at async PluginManager.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:592:7)\n    at async Serverless.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/serverless.js:169:5)\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:366:15\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:235:28\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/setup-run-serverless-fixtures-engine.js:33:22\n    at async Context.<anonymous> (test/integration/deploy.test.js:22:9)
2022-02-25T05:20:38.900Z i mocha flushing previously gathered logs...
2022-02-25T05:20:38.290Z * mocha client deploy integration test > us-west-2
2022-02-25T05:20:38.290Z * mocha client deploy integration test > us-west-2 > should deploy site sucessfully
2022-02-25T05:20:38.306Z i serverless:test setup basic fixture at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/tmpdirs-serverless/cd1c/b001d6
2022-02-25T05:20:38.313Z i serverless To ensure safe major version upgrades ensure "frameworkVersion" setting in service configuration (recommended setup: "frameworkVersion: ^3.3.0")

2022-02-25T05:20:38.313Z * serverless:lifecycle:command:register config
2022-02-25T05:20:38.313Z * serverless:lifecycle:command:register config:credentials
2022-02-25T05:20:38.314Z * serverless:lifecycle:command:register create
2022-02-25T05:20:38.314Z * serverless:lifecycle:command:register install
2022-02-25T05:20:38.314Z * serverless:lifecycle:command:register package
2022-02-25T05:20:38.314Z * serverless:lifecycle:command:register package:function
2022-02-25T05:20:38.314Z * serverless:lifecycle:command:register deploy
2022-02-25T05:20:38.314Z * serverless:lifecycle:command:register deploy:function
2022-02-25T05:20:38.314Z * serverless:lifecycle:command:register deploy:list
2022-02-25T05:20:38.314Z * serverless:lifecycle:command:register deploy:list:functions
2022-02-25T05:20:38.315Z * serverless:lifecycle:command:register invoke
2022-02-25T05:20:38.315Z * serverless:lifecycle:command:register invoke:local
2022-02-25T05:20:38.315Z * serverless:lifecycle:command:register info
2022-02-25T05:20:38.315Z * serverless:lifecycle:command:register logs
2022-02-25T05:20:38.315Z * serverless:lifecycle:command:register metrics
2022-02-25T05:20:38.315Z * serverless:lifecycle:command:register print
2022-02-25T05:20:38.315Z * serverless:lifecycle:command:register remove
2022-02-25T05:20:38.315Z * serverless:lifecycle:command:register rollback
2022-02-25T05:20:38.315Z * serverless:lifecycle:command:register rollback:function
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register slstats
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register plugin:list
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register plugin:search
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register config
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register config:credentials
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register aws:common
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register aws:common:validate
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register aws:common:cleanupTempDir
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register aws:common:moveArtifactsToPackage
2022-02-25T05:20:38.316Z * serverless:lifecycle:command:register aws:common:moveArtifactsToTemp
2022-02-25T05:20:38.317Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:38.317Z * serverless:lifecycle:command:register aws:package
2022-02-25T05:20:38.317Z * serverless:lifecycle:command:register aws:package:finalize
2022-02-25T05:20:38.317Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:38.317Z * serverless:lifecycle:command:register aws:deploy
2022-02-25T05:20:38.317Z * serverless:lifecycle:command:register aws:deploy:deploy
2022-02-25T05:20:38.317Z * serverless:lifecycle:command:register aws:deploy:finalize
2022-02-25T05:20:38.317Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:38.317Z * serverless:lifecycle:command:register aws:info
2022-02-25T05:20:38.317Z * serverless:lifecycle:command:register upgrade
2022-02-25T05:20:38.318Z * serverless:lifecycle:command:register uninstall
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register client
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register client:deploy
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register client:remove
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register login
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register logout
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register generate-event
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register test
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register dashboard
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register output
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register output:get
2022-02-25T05:20:38.319Z * serverless:lifecycle:command:register output:list
2022-02-25T05:20:38.320Z * serverless:lifecycle:command:register param
2022-02-25T05:20:38.320Z * serverless:lifecycle:command:register param:get
2022-02-25T05:20:38.320Z * serverless:lifecycle:command:register param:list
2022-02-25T05:20:38.419Z * serverless:lifecycle:command:invoke Invoke client:deploy
2022-02-25T05:20:38.419Z i serverless:plugin-legacy This deployment will:
2022-02-25T05:20:38.419Z i serverless:plugin-legacy - Upload all files from 'client/dist' to bucket 'sls-finch-test-stndrd-usw2-fer' undefined
2022-02-25T05:20:38.419Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-usw2-fer' configuration
2022-02-25T05:20:38.419Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-usw2-fer' bucket policy
2022-02-25T05:20:38.419Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-usw2-fer' CORS policy
2022-02-25T05:20:38.419Z i serverless:plugin-legacy Looking for bucket...
2022-02-25T05:20:38.895Z * serverless:aws [AWS s3 403 0.475s 0 retries] listBuckets({})
2022-02-25T05:20:38.896Z * serverless InvalidAccessKeyId: The AWS Access Key Id you provided does not exist in our records.
    at Request.extractError (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/services/s3.js:710:35)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at callNextListener (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:96:12)
    at IncomingMessage.onEnd (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/event_listeners.js:335:13)
    at IncomingMessage.emit (events.js:412:35)
    at IncomingMessage.emit (domain.js:475:12)
    at endReadableNT (internal/streams/readable.js:1334:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
----------------------------------------------------------------------------------------------------
    custom headers
2022-02-25T05:20:39.038Z i serverless:plugin-legacy This deployment will:
2022-02-25T05:20:39.038Z i serverless:plugin-legacy - Upload all files from 'client/dist' to bucket 'sls-finch-test-stndrd-use1-v3' undefined
2022-02-25T05:20:39.038Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' configuration
2022-02-25T05:20:39.038Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' bucket policy
2022-02-25T05:20:39.038Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' CORS policy
2022-02-25T05:20:39.039Z i serverless:plugin-legacy Looking for bucket...
      3) should be applied
2022-02-25T05:20:39.187Z × mocha test fail ServerlessError: ServerlessError: The AWS Access Key Id you provided does not exist in our records.\n    at /Users/shereef.marzouk/Dev/serverless-finch/index.js:314:31\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at async PluginManager.runHooks (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:522:35)\n    at async PluginManager.invoke (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:552:9)\n    at async PluginManager.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:592:7)\n    at async Serverless.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/serverless.js:169:5)\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:366:15\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:235:28\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/setup-run-serverless-fixtures-engine.js:33:22\n    at async Context.<anonymous> (test/integration/deploy.test.js:33:7)
2022-02-25T05:20:39.188Z i mocha flushing previously gathered logs...
2022-02-25T05:20:38.902Z * mocha client deploy integration test > custom headers
2022-02-25T05:20:38.902Z * mocha client deploy integration test > custom headers > should be applied
2022-02-25T05:20:38.917Z i serverless:test setup custom-headers fixture at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/tmpdirs-serverless/cd1c/52a3e8
2022-02-25T05:20:38.923Z i serverless To ensure safe major version upgrades ensure "frameworkVersion" setting in service configuration (recommended setup: "frameworkVersion: ^3.3.0")

2022-02-25T05:20:38.923Z * serverless:lifecycle:command:register config
2022-02-25T05:20:38.923Z * serverless:lifecycle:command:register config:credentials
2022-02-25T05:20:38.924Z * serverless:lifecycle:command:register create
2022-02-25T05:20:38.924Z * serverless:lifecycle:command:register install
2022-02-25T05:20:38.924Z * serverless:lifecycle:command:register package
2022-02-25T05:20:38.924Z * serverless:lifecycle:command:register package:function
2022-02-25T05:20:38.924Z * serverless:lifecycle:command:register deploy
2022-02-25T05:20:38.924Z * serverless:lifecycle:command:register deploy:function
2022-02-25T05:20:38.924Z * serverless:lifecycle:command:register deploy:list
2022-02-25T05:20:38.924Z * serverless:lifecycle:command:register deploy:list:functions
2022-02-25T05:20:38.925Z * serverless:lifecycle:command:register invoke
2022-02-25T05:20:38.925Z * serverless:lifecycle:command:register invoke:local
2022-02-25T05:20:38.925Z * serverless:lifecycle:command:register info
2022-02-25T05:20:38.925Z * serverless:lifecycle:command:register logs
2022-02-25T05:20:38.926Z * serverless:lifecycle:command:register metrics
2022-02-25T05:20:38.926Z * serverless:lifecycle:command:register print
2022-02-25T05:20:38.926Z * serverless:lifecycle:command:register remove
2022-02-25T05:20:38.926Z * serverless:lifecycle:command:register rollback
2022-02-25T05:20:38.926Z * serverless:lifecycle:command:register rollback:function
2022-02-25T05:20:38.926Z * serverless:lifecycle:command:register slstats
2022-02-25T05:20:38.926Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:38.926Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:38.926Z * serverless:lifecycle:command:register plugin:list
2022-02-25T05:20:38.927Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:38.927Z * serverless:lifecycle:command:register plugin:search
2022-02-25T05:20:38.927Z * serverless:lifecycle:command:register config
2022-02-25T05:20:38.927Z * serverless:lifecycle:command:register config:credentials
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws:common
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws:common:validate
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws:common:cleanupTempDir
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws:common:moveArtifactsToPackage
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws:common:moveArtifactsToTemp
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws:package
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws:package:finalize
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws:deploy
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws:deploy:deploy
2022-02-25T05:20:38.928Z * serverless:lifecycle:command:register aws:deploy:finalize
2022-02-25T05:20:38.929Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:38.929Z * serverless:lifecycle:command:register aws:info
2022-02-25T05:20:38.929Z * serverless:lifecycle:command:register upgrade
2022-02-25T05:20:38.930Z * serverless:lifecycle:command:register uninstall
2022-02-25T05:20:38.930Z * serverless:lifecycle:command:register client
2022-02-25T05:20:38.930Z * serverless:lifecycle:command:register client:deploy
2022-02-25T05:20:38.930Z * serverless:lifecycle:command:register client:remove
2022-02-25T05:20:38.930Z * serverless:lifecycle:command:register login
2022-02-25T05:20:38.931Z * serverless:lifecycle:command:register logout
2022-02-25T05:20:38.931Z * serverless:lifecycle:command:register generate-event
2022-02-25T05:20:38.931Z * serverless:lifecycle:command:register test
2022-02-25T05:20:38.931Z * serverless:lifecycle:command:register dashboard
2022-02-25T05:20:38.931Z * serverless:lifecycle:command:register output
2022-02-25T05:20:38.931Z * serverless:lifecycle:command:register output:get
2022-02-25T05:20:38.931Z * serverless:lifecycle:command:register output:list
2022-02-25T05:20:38.931Z * serverless:lifecycle:command:register param
2022-02-25T05:20:38.931Z * serverless:lifecycle:command:register param:get
2022-02-25T05:20:38.931Z * serverless:lifecycle:command:register param:list
2022-02-25T05:20:39.037Z * serverless:lifecycle:command:invoke Invoke client:deploy
2022-02-25T05:20:39.038Z i serverless:plugin-legacy This deployment will:
2022-02-25T05:20:39.038Z i serverless:plugin-legacy - Upload all files from 'client/dist' to bucket 'sls-finch-test-stndrd-use1-v3' undefined
2022-02-25T05:20:39.038Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' configuration
2022-02-25T05:20:39.038Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' bucket policy
2022-02-25T05:20:39.038Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' CORS policy
2022-02-25T05:20:39.039Z i serverless:plugin-legacy Looking for bucket...
2022-02-25T05:20:39.186Z * serverless:aws [AWS s3 403 0.146s 0 retries] listBuckets({})
2022-02-25T05:20:39.187Z * serverless InvalidAccessKeyId: The AWS Access Key Id you provided does not exist in our records.
    at Request.extractError (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/services/s3.js:710:35)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at callNextListener (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:96:12)
    at IncomingMessage.onEnd (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/event_listeners.js:335:13)
    at IncomingMessage.emit (events.js:412:35)
    at IncomingMessage.emit (domain.js:475:12)
    at endReadableNT (internal/streams/readable.js:1334:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
----------------------------------------------------------------------------------------------------
    routing rules redirect
2022-02-25T05:20:39.300Z i serverless:plugin-legacy This deployment will:
2022-02-25T05:20:39.301Z i serverless:plugin-legacy - Upload all files from 'client/dist' to bucket 'sls-finch-test-stndrd-use1-v3' undefined
2022-02-25T05:20:39.301Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' configuration
2022-02-25T05:20:39.301Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' bucket policy
2022-02-25T05:20:39.301Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' CORS policy
2022-02-25T05:20:39.301Z i serverless:plugin-legacy Looking for bucket...
      4) should cause a 404 request to be redirected to the index
2022-02-25T05:20:39.450Z × mocha test fail ServerlessError: ServerlessError: The AWS Access Key Id you provided does not exist in our records.\n    at /Users/shereef.marzouk/Dev/serverless-finch/index.js:314:31\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at async PluginManager.runHooks (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:522:35)\n    at async PluginManager.invoke (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:552:9)\n    at async PluginManager.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:592:7)\n    at async Serverless.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/serverless.js:169:5)\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:366:15\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:235:28\n    at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/setup-run-serverless-fixtures-engine.js:33:22\n    at async Context.<anonymous> (test/integration/deploy.test.js:43:7)
2022-02-25T05:20:39.450Z i mocha flushing previously gathered logs...
2022-02-25T05:20:39.189Z * mocha client deploy integration test > routing rules redirect
2022-02-25T05:20:39.189Z * mocha client deploy integration test > routing rules redirect > should cause a 404 request to be redirected to the index
2022-02-25T05:20:39.199Z i serverless:test setup routing-rules-redirect fixture at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/tmpdirs-serverless/cd1c/f0ea65
2022-02-25T05:20:39.203Z i serverless To ensure safe major version upgrades ensure "frameworkVersion" setting in service configuration (recommended setup: "frameworkVersion: ^3.3.0")

2022-02-25T05:20:39.203Z * serverless:lifecycle:command:register config
2022-02-25T05:20:39.203Z * serverless:lifecycle:command:register config:credentials
2022-02-25T05:20:39.204Z * serverless:lifecycle:command:register create
2022-02-25T05:20:39.204Z * serverless:lifecycle:command:register install
2022-02-25T05:20:39.204Z * serverless:lifecycle:command:register package
2022-02-25T05:20:39.204Z * serverless:lifecycle:command:register package:function
2022-02-25T05:20:39.204Z * serverless:lifecycle:command:register deploy
2022-02-25T05:20:39.204Z * serverless:lifecycle:command:register deploy:function
2022-02-25T05:20:39.204Z * serverless:lifecycle:command:register deploy:list
2022-02-25T05:20:39.204Z * serverless:lifecycle:command:register deploy:list:functions
2022-02-25T05:20:39.204Z * serverless:lifecycle:command:register invoke
2022-02-25T05:20:39.204Z * serverless:lifecycle:command:register invoke:local
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register info
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register logs
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register metrics
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register print
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register remove
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register rollback
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register rollback:function
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register slstats
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:39.205Z * serverless:lifecycle:command:register plugin:list
2022-02-25T05:20:39.206Z * serverless:lifecycle:command:register plugin
2022-02-25T05:20:39.206Z * serverless:lifecycle:command:register plugin:search
2022-02-25T05:20:39.206Z * serverless:lifecycle:command:register config
2022-02-25T05:20:39.206Z * serverless:lifecycle:command:register config:credentials
2022-02-25T05:20:39.208Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:39.209Z * serverless:lifecycle:command:register aws:common
2022-02-25T05:20:39.209Z * serverless:lifecycle:command:register aws:common:validate
2022-02-25T05:20:39.209Z * serverless:lifecycle:command:register aws:common:cleanupTempDir
2022-02-25T05:20:39.209Z * serverless:lifecycle:command:register aws:common:moveArtifactsToPackage
2022-02-25T05:20:39.209Z * serverless:lifecycle:command:register aws:common:moveArtifactsToTemp
2022-02-25T05:20:39.210Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:39.210Z * serverless:lifecycle:command:register aws:package
2022-02-25T05:20:39.210Z * serverless:lifecycle:command:register aws:package:finalize
2022-02-25T05:20:39.210Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:39.210Z * serverless:lifecycle:command:register aws:deploy
2022-02-25T05:20:39.210Z * serverless:lifecycle:command:register aws:deploy:deploy
2022-02-25T05:20:39.210Z * serverless:lifecycle:command:register aws:deploy:finalize
2022-02-25T05:20:39.210Z * serverless:lifecycle:command:register aws
2022-02-25T05:20:39.210Z * serverless:lifecycle:command:register aws:info
2022-02-25T05:20:39.211Z * serverless:lifecycle:command:register upgrade
2022-02-25T05:20:39.211Z * serverless:lifecycle:command:register uninstall
2022-02-25T05:20:39.211Z * serverless:lifecycle:command:register client
2022-02-25T05:20:39.211Z * serverless:lifecycle:command:register client:deploy
2022-02-25T05:20:39.211Z * serverless:lifecycle:command:register client:remove
2022-02-25T05:20:39.212Z * serverless:lifecycle:command:register login
2022-02-25T05:20:39.212Z * serverless:lifecycle:command:register logout
2022-02-25T05:20:39.212Z * serverless:lifecycle:command:register generate-event
2022-02-25T05:20:39.212Z * serverless:lifecycle:command:register test
2022-02-25T05:20:39.213Z * serverless:lifecycle:command:register dashboard
2022-02-25T05:20:39.213Z * serverless:lifecycle:command:register output
2022-02-25T05:20:39.213Z * serverless:lifecycle:command:register output:get
2022-02-25T05:20:39.213Z * serverless:lifecycle:command:register output:list
2022-02-25T05:20:39.213Z * serverless:lifecycle:command:register param
2022-02-25T05:20:39.213Z * serverless:lifecycle:command:register param:get
2022-02-25T05:20:39.213Z * serverless:lifecycle:command:register param:list
2022-02-25T05:20:39.300Z * serverless:lifecycle:command:invoke Invoke client:deploy
2022-02-25T05:20:39.300Z i serverless:plugin-legacy This deployment will:
2022-02-25T05:20:39.301Z i serverless:plugin-legacy - Upload all files from 'client/dist' to bucket 'sls-finch-test-stndrd-use1-v3' undefined
2022-02-25T05:20:39.301Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' configuration
2022-02-25T05:20:39.301Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' bucket policy
2022-02-25T05:20:39.301Z i serverless:plugin-legacy - Set (and overwrite) bucket 'sls-finch-test-stndrd-use1-v3' CORS policy
2022-02-25T05:20:39.301Z i serverless:plugin-legacy Looking for bucket...
2022-02-25T05:20:39.449Z * serverless:aws [AWS s3 403 0.148s 0 retries] listBuckets({})
2022-02-25T05:20:39.449Z * serverless InvalidAccessKeyId: The AWS Access Key Id you provided does not exist in our records.
    at Request.extractError (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/services/s3.js:710:35)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at callNextListener (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/sequential_executor.js:96:12)
    at IncomingMessage.onEnd (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/node_modules/aws-sdk/lib/event_listeners.js:335:13)
    at IncomingMessage.emit (events.js:412:35)
    at IncomingMessage.emit (domain.js:475:12)
    at endReadableNT (internal/streams/readable.js:1334:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
----------------------------------------------------------------------------------------------------


  0 passing (1m)
  4 failing

  1) client deploy integration test
       us-east-1
         should deploy site sucessfully:
     ServerlessError: ServerlessError: The AWS Access Key Id you provided does not exist in our records.
      at /Users/shereef.marzouk/Dev/serverless-finch/index.js:314:31
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at async PluginManager.runHooks (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:522:35)
      at async PluginManager.invoke (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:552:9)
      at async PluginManager.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:592:7)
      at async Serverless.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/serverless.js:169:5)
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:366:15
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:235:28
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/setup-run-serverless-fixtures-engine.js:33:22
      at async Context.<anonymous> (test/integration/deploy.test.js:22:9)

  2) client deploy integration test
       us-west-2
         should deploy site sucessfully:
     ServerlessError: ServerlessError: The AWS Access Key Id you provided does not exist in our records.
      at /Users/shereef.marzouk/Dev/serverless-finch/index.js:314:31
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at async PluginManager.runHooks (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:522:35)
      at async PluginManager.invoke (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:552:9)
      at async PluginManager.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:592:7)
      at async Serverless.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/serverless.js:169:5)
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:366:15
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:235:28
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/setup-run-serverless-fixtures-engine.js:33:22
      at async Context.<anonymous> (test/integration/deploy.test.js:22:9)

  3) client deploy integration test
       custom headers
         should be applied:
     ServerlessError: ServerlessError: The AWS Access Key Id you provided does not exist in our records.
      at /Users/shereef.marzouk/Dev/serverless-finch/index.js:314:31
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at async PluginManager.runHooks (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:522:35)
      at async PluginManager.invoke (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:552:9)
      at async PluginManager.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:592:7)
      at async Serverless.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/serverless.js:169:5)
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:366:15
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:235:28
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/setup-run-serverless-fixtures-engine.js:33:22
      at async Context.<anonymous> (test/integration/deploy.test.js:33:7)

  4) client deploy integration test
       routing rules redirect
         should cause a 404 request to be redirected to the index:
     ServerlessError: ServerlessError: The AWS Access Key Id you provided does not exist in our records.
      at /Users/shereef.marzouk/Dev/serverless-finch/index.js:314:31
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at async PluginManager.runHooks (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:522:35)
      at async PluginManager.invoke (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:552:9)
      at async PluginManager.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/classes/plugin-manager.js:592:7)
      at async Serverless.run (/private/var/folders/_f/ttmy11913fv62d6sz_6hkjlm0000gq/T/serverless-finch-plugin-test-9af3/lib/serverless.js:169:5)
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:366:15
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/run-serverless.js:235:28
      at async /Users/shereef.marzouk/Dev/serverless-finch/node_modules/@serverless/test/setup-run-serverless-fixtures-engine.js:33:22
      at async Context.<anonymous> (test/integration/deploy.test.js:43:7)



`node node_modules/.bin/_mocha test/integration/deploy.test.js` Exited with code 4

test/integration/deploy.test.js failed


test/integration/deploy.test.js failed
Shmac:serverless-finch shereef.marzouk$ 
```

What am I doing wrong ?

Reply:
FYI the AWS account i used I have full admin access on

```
aws sts get-caller-identity

{
    "UserId": "*********:shereef.marzouk",
    "Account": "************",
    "Arn": "arn:aws:sts::************:assumed-role/AWSReservedSSO_AdministratorAccess_8*****2/shereef.marzouk"
}
```

Reply:
Thanks for the review. I removed the second commit from this PR, I totally agree it belongs separately. However I couldn't figure out how to make the other PR "on top" of this one.

For the integration test issue I'll comment on #143

Reply:
You are right. In order to make that PR depend on this one we'd have to make it point to this PR's branch which is an external branch which would not link this PR to the upstream repo (this repo).

I like how you set it up right now we just need to know that #143 shouldn't be merged until #138 is merged